vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java

```diff
@@ -21,11 +33,17 @@
     @Autowired
     private ApplicationContext applicationContext;
 
+    @Autowired
```

Comment:
This is broken since only by updating to flow 8.0 and vaadin-spring 18.0 in a SpringBoot based application with no SecurityConfigurerAdapter, it cannot start. 
This is because `RequestUtil` instance is being created by `SpringBootAutoConfiguration`, while its dependency, `AccessAnnotationChecker`, has not been created as there is no `WebSecurityConfigurerAdapter` present in the classpath (and `SpringSecurityAutoConfiguration` has a @ConditionalOnClass on that). 
Changing `@Autowired` to `@Autowired(required = false)` would alleviate the problem temporarily, but there might be a better way.

Reply:
Let's see first that we have a test that fails because of this

Reply:
This PR is dependant on https://github.com/vaadin/flow/pull/10863 to be merged first. 
Even after that, an existing spring boot application cannot start to pass any test.

Reply:
`RequestUtil` should only be needed for Spring Security so let's move the registration

Reply:
Fixed